### PR TITLE
add model for inferring the type of function callback arguments

### PR DESCRIPTION
**Pull request by i582**
This model allows to describe the types of arguments for callback functions depending on the arguments of the calling function.
[WI-54960](https://youtrack.jetbrains.com/issue/WI-54960)

### CODE REVIEW

File: meta/.phpstorm.meta.php [hunk 1]
```diff
@@ -653,6 +653,43 @@ function exitPoint($functionReference) {
...
+     * @param mixed $args
+     * @see arg()
+     * @return mixed
+     */
+    function callback(...$args) {
+      return "callback ${$args}";
+    }
+
+    /**
+     * argument type #$argNum of the caller callback
```
Comment (YuriyNasretdinov) [inline meta/.phpstorm.meta.php:677]: This needs to be a type of an _element_, not the type of the argument (argument to array_map is array, and the callbacks accept an element of an array)

**Review by YuriyNasretdinov — CHANGES_REQUESTED** [review]

File: meta/.phpstorm.meta.php (hunk 1 above)
Comment (i582) [reply meta/.phpstorm.meta.php]: Yes, it is, thanks.

**Comment by b1rdex** [comment]
I've already added a comment on the issue page and saying that IMO it's better to get support for generics (https://youtrack.jetbrains.com/issue/WI-47158) rather than create a new proprietary thing.

**Comment by slapoguzov** [comment]
Thanks for your contribution. However, this is a new syntax for `.phpstorm.meta.php` and we have to consider these changes more detailed. Please,[ create a new ticket at our tracker](https://youtrack.jetbrains.com/newIssue?project=WI) and describe your suggestions there. You also can attach this mr to the ticket.

**Pull request closed without merging**